asteroid/utils/generic_utils.py

```diff
@@ -75,12 +75,34 @@ def get_wav_random_start_stop(signal_len, desired_len=4 * 8000):
     Returns:
         tuple: random start integer, stop integer.
     """
-    if signal_len == desired_len or desired_len is None:
-        rand_start = 0
-    else:
-        rand_start = np.random.randint(0, signal_len - desired_len)
     if desired_len is None:
-        stop = None
-    else:
-        stop = rand_start + desired_len
+        return 0, signal_len
+    rand_start = np.random.randint(0, max(1, signal_len - desired_len))
+    stop = min(signal_len, rand_start + desired_len)
     return rand_start, stop
+
+
+def unet_decoder_args(encoders, *, skip_connections):
```

Comment:
Maybe it's specific, I don't know... I have seen it many times, e.g. "Singing Voice Separation with Deep U-Net Convolutional Networks", Spleeter, DCUNet, DCCRNet, ... granted, those are all variations of the same u-net structure. But I think even in the original u-net paper they use this structure.

`skip_connections`: The `*` forces the following arguments to be keyword-only arguments. It's just a way to force people to call it like `unet_decoder_args(..., skip_connections=True)` rather than `unet_decoder_args(..., True)`, which is less readable code. So just a way to force people to write better code. I always do it for arguments that don't have an obvious meaning at the caller site. But I don't have strong opinions about it, so if you don't like it I can remove it.

Reply:
- Ok to keep this code in utils then, thanks.
- Ok for that. I would maybe add `True` as a default value, WDYT?